cli/export.go

```diff
@@ -531,26 +571,96 @@ func runImport(args *docopt.Args, client controller.Client) error {
 		}
 	}
 
-	uploadSlug := release != nil && imageArtifact != nil && slug != nil
-
-	if uploadSlug {
-		// Use current slugrunner as the artifact
+	if legacySlug != nil {
+		// Use current slugbuilder to convert the legacy slug to a
+		// Flynn squashfs image
 		gitreceiveRelease, err := client.GetAppRelease("gitreceive")
 		if err != nil {
 			return fmt.Errorf("unable to retrieve gitreceive release: %s", err)
 		}
-		if id, ok := gitreceiveRelease.Env["SLUGRUNNER_IMAGE_ID"]; ok {
-			imageArtifact, err = client.GetArtifact(id)
-			if err != nil {
-				return fmt.Errorf("unable to get slugrunner image artifact: %s", err)
+		slugBuilderID, ok := gitreceiveRelease.Env["SLUGBUILDER_IMAGE_ID"]
+		if !ok {
+			return fmt.Errorf("gitreceive env missing slug builder image")
+		}
+		slugRunnerID, ok := gitreceiveRelease.Env["SLUGRUNNER_IMAGE_ID"]
+		if !ok {
+			return fmt.Errorf("gitreceive env missing slug runner image")
+		}
+
+		slugImageID := random.UUID()
+		config := runConfig{
+			App:        app.ID,
+			Release:    gitreceiveRelease.ID,
+			ReleaseEnv: true,
+			Artifacts:  []string{slugBuilderID},
+			DisableLog: true,
+			Args:       []string{"/bin/convert-legacy-slug.sh"},
```

Comment:
What will happen if the import command is called against a cluster that doesn't have this code?